[VSC-42] Add a test for local and nested modules

heitor-lassarote opened the pull request:
Problem: Now that we have better support for recovering the typed AST for dot expressions in the compiler, we should also add a test for it in the extension.

Solution: Add a test checking that we can complete from a local module and another that we can complete from a nested module. While testing, I found that we don't properly filter items based on their prefix, so I added this case for completions from definitions.

heitor-lassarote committed:
[VSC-42] Add a test for local and nested modules

Problem: Now that we have better support for recovering the typed AST
for dot expressions in the compiler, we should also add a test for it in
the extension.

Solution: Add a test checking that we can complete from a local module
and another that we can complete from a nested module. While testing, I
found that we don't properly filter items based on their prefix, so I
added this case for completions from definitions.

alexd1971 reviewed (approved):
Code looks good. Only one question regarding this:

> we don't properly filter items based on their prefix

I see we can have two approaches:
1. Filter completion items on the server side and return filtered list.
2. Return full completion list and filter on the client side (it seems to be done automatically).

I'm not sure which one is better? If we'd use some complex filtering on the server side it would be justified. But we use just `startsWith`, what I think is done on the client side as well.

heitor-lassarote committed:
[VSC-42] Do not filter items server side

Problem: According to the LSP specification, the server should not
filter items unless we send `isIncomplete` is `true`. However, this
would cause the client to recompute the list on every keystroke, and
since we rely on `findDefinitions`, which is slow, we don't want this.

Solution: Do not filter items based on a prefix. To check for the items,
order, and continuity in a subarray, we had to write a custom Jest
matcher. Set `isIncomplete` as `false` since we now send the complete
list.
